src/printer.cpp

```diff
@@ -127,12 +135,32 @@ std::string printConnections(const ComponentMap &componentMap, const VariableMap
     return connections;
 }
 
-std::string printMath(const std::string &math)
+std::string Printer::PrinterImpl::printMath(const std::string &math)
 {
     static const std::regex before(">[\\s\n\t]*");
     static const std::regex after("[\\s\n\t]*<");
-    auto temp = std::regex_replace(math, before, ">");
-    return std::regex_replace(temp, after, "<");
+    static const std::string xmlDeclaration = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>";
```

Comment:
are we restricted to XML 1.0 and UTF-8 encoding? I don't think we are...and if we are going to do a string match replacement then probably better to do a regex replace to handle XML 1.1 and other encodings (or no encoding) and the possible `standalone` attribute.

Reply:
`xmlDeclaration` is used after we have done a pretty print, which seems to be using UTF-8 (see `XmlDoc::prettyPrint()`), so it looks like it might be fine?

Reply:
This is the relevant code for the XML declaration:
```
	/*
	 * Save the XML declaration
	 */
	if ((ctxt->options & XML_SAVE_NO_DECL) == 0) {
	    xmlOutputBufferWrite(buf, 14, "<?xml version=");
	    if (cur->version != NULL) 
		xmlBufferWriteQuotedString(buf->buffer, cur->version);
	    else
		xmlOutputBufferWrite(buf, 5, "\"1.0\"");
	    if (encoding != NULL) {
		xmlOutputBufferWrite(buf, 10, " encoding=");
		xmlBufferWriteQuotedString(buf->buffer, (xmlChar *) encoding);
	    }
	    switch (cur->standalone) {
		case 0:
		    xmlOutputBufferWrite(buf, 16, " standalone=\"no\"");
		    break;
		case 1:
		    xmlOutputBufferWrite(buf, 17, " standalone=\"yes\"");
		    break;
	    }
	    xmlOutputBufferWrite(buf, 3, "?>\n");
	}
```
This has to be the first thing you see in the text so we could add a regex around this.

Reply:
wait?! the `prettyPrint` adds the XML declaration in? so can't we just expose the `XML_SAVE_NO_DECL` option and not manipulate the string afterwards?

Reply:
I can't seem to find a way to set the `XML_SAVE_NO_DECL` option, if you can tell me how I will do it.

Reply:
can't see an obvious way using the `xmlDocDumpFormatMemoryEnc` approach....would need to switch to the `xmlsave` approach. regex it is :)